lib/ansible/modules/web_infrastructure/jenkins_plugin.py

```diff
@@ -448,31 +450,39 @@ def install(self):
                         self.params['name'],
                         self.params['version']))
 
+            sha1sum_old = base64.b64encode(sha1_old.digest())
             if (
                     self.params['updates_expiration'] == 0 or
                     self.params['version'] not in [None, 'latest'] or
-                    md5sum_old is None):
+                    sha1sum_old is None):
 
                 # Download the plugin file directly
                 r = self._download_plugin(plugin_url)
 
                 # Write downloaded plugin into file if checksums don't match
-                if md5sum_old is None:
+                if sha1sum_old is None:
                     # No previously installed plugin
                     if not self.module.check_mode:
                         self._write_file(plugin_file, r)
 
                     changed = True
                 else:
-                    # Get data for the MD5
+                    # Get data for the SHA1
                     data = r.read()
 
                     # Make new checksum
-                    md5sum_new = hashlib.md5(data).hexdigest()
+                    try:
+                        sha1_new = hashlib.sha1(data)
+                    except Exception as err:
+                        self.module.fail_json(
+                            msg="Cannot calculate SHA1 of the downloaded plugin.",
+                            details="SHA1 calculation failed with: {0}".format(err)
+                        )
+                    sha1sum_new = base64.b64encode(sha1_new.digest())
```

Comment:
Under what condition would this exception handler even run?

Reply:
I would guess that it could only throw an exception, in the scenario, where there is not access to the plugin_url, and thereby the `data` will be null, you could very well argue such a check should b around `r.read()` and not around the later usage. Let me implement all the suggestion and come back with a new implementation. I will also merge master in